analytics/analytics.go

```diff
@@ -0,0 +1,258 @@
+package analytics
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/alecthomas/kong"
+	"github.com/denisbrodbeck/machineid"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+var (
+	// HeapBaseURI is the location of Heap API endpoint
+	HeapBaseURI = "https://heapanalytics.com"
+	// HeapAppID identifies what Heap App events are recorded against
+	HeapAppID = "4248790180"
```

Comment:
Doesn't this come from the build?

Reply:
Almost! 

The `HeapAppID` in the source code is the ID of our development environment. 

During the build, we override it with the production ID. 

The rationale here is to keep analytics generated during dev/test separate to production usage.

Reply:
Added a comment to this line in 8069d33307a83bb57987606675aa8597f4ef631d to make it clearer what's happening.